### PR TITLE
Run livenessprobe as nonroot

### PR DESCRIPTION
**Pull request by prankulmahajan**
**This PR aims to do the following,**

1. In deployment set `runAsGroup` for all containers in nodeServer and controller. Currently, though the users are non-root, the group is still being used as root.
2. In deployment run `livenessprobe` sidecar as non-root user/group.

To achieve the 2nd point, had to change file permissions of CSI socket created by node server.

- Default permissions of csi socket inside node server -- (User 0, Group 0, Permissions 755) -- Only the root user has permissions to access the socket
- The socket file should have **read and write** permissions for access. i.e min access required is 600/660 depending on who is accessing the socket file

This PR changes the group of the csi socket from root to non-root(set 2121 in deployment configMap) and update permissions to 660. Setting "660" allows the root user and the non-root group to access the socket as expected.

### CODE REVIEW

**Comment by prankulmahajan** [comment]
## Test Results 

- [x] Setup -- IKS 1.30
```
❯ kubectl get nodes
NAME          STATUS   ROLES    AGE   VERSION
10.240.0.85   Ready    <none>   2d    v1.30.4+IKS
10.240.0.86   Ready    <none>   2d    v1.30.4+IKS
```
- [ ] Resource status
```
❯ kubectl get pods -n kube-system -o wide | grep block
ibm-vpc-block-csi-controller-55cf4685b4-tq7mp         7/7     Running   0          5m18s   172.17.109.19    10.240.0.85   <none>           <none>
ibm-vpc-block-csi-node-l5flp                          4/4     Running   0          5m18s   172.17.112.156   10.240.0.86   <none>           <none>
ibm-vpc-block-csi-node-pdr9h                          4/4     Running   0          5m18s   172.17.109.20    10.240.0.85   <none>           <none>
```
- [ ] Check if csi socket is retained after resource deletion -- socket got deleted 
```
root@test-cr624av200pnjn76o2i0-iks130-default-0000018a:/var/lib/kubelet/plugins# ls
kubernetes.io  vpc.block.csi.ibm.io  vpc.file.csi.ibm.io
root@test-cr624av200pnjn76o2i0-iks130-default-0000018a:/var/lib/kubelet/plugins# cd vpc.block.csi.ibm.io/
root@test-cr624av200pnjn76o2i0-iks130-default-0000018a:/var/lib/kubelet/plugins/vpc.block.csi.ibm.io# ls -la
total 8
drwxr-xr-x 2 root root 4096 Aug 28 07:34 .
drwxr-x--- 5 root root 4096 Aug 26 14:40 ..
root@test-cr624av200pnjn76o2i0-iks130-default-0000018a:/var/lib/kubelet/plugins/vpc.block.csi.ibm.io#
```
- [ ] Apply resources using custom image and check status + Exec into node and check csi socket permissions
```
❯ kubectl get pods -n kube-system -o wide | grep block
ibm-vpc-block-csi-controller-55cf4685b4-zsg2f         7/7     Running   0          2m52s   172.17.109.22    10.240.0.85   <none>           <none>
ibm-vpc-block-csi-node-f4n4f                          4/4     Running   0          2m53s   172.17.112.157   10.240.0.86   <none>           <none>
ibm-vpc-block-csi-node-wv7w7                          4/4     Running   0          2m53s   172.17.109.21    10.240.0.85   <none>           <none>


❯ runon 10.240.0.85 command bash
groups: cannot find name for group ID 11
To run a command as administrator (user "root"), use "sudo <command>".
See "man sudo_root" for details.

root@test-cr624av200pnjn76o2i0-iks130-default-0000018a:/# cd /var/lib/kubelet/plugins/vpc.block.csi.ibm.io/
root@test-cr624av200pnjn76o2i0-iks130-default-0000018a:/var/lib/kubelet/plugins/vpc.block.csi.ibm.io# ls
csi.sock
root@test-cr624av200pnjn76o2i0-iks130-default-0000018a:/var/lib/kubelet/plugins/vpc.block.csi.ibm.io# ls -la
total 8
drwxr-xr-x 2 root root 4096 Aug 28 07:40 .
drwxr-x--- 5 root root 4096 Aug 26 14:40 ..
srw-rw---- 1 root 2121    0 Aug 28 07:40 csi.sock
root@test-cr624av200pnjn76o2i0-iks130-default-0000018a:/var/lib/kubelet/plugins/vpc.block.csi.ibm.io#
```
- [ ] Create PVC and attach it to a pod
```
❯ kubectl get pvc -w
NAME           STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS                VOLUMEATTRIBUTESCLASS   AGE
my-pvc-block   Bound    pvc-621d1283-e1fc-4d0c-ba63-a8533a2f700c   10Gi       RWO            ibmc-vpc-block-5iops-tier   <unset>                 22s
^C%

❯ kubectl get pods -o wide
NAME                                 READY   STATUS    RESTARTS   AGE    IP              NODE          NOMINATED NODE   READINESS GATES
auto-expansion-pod-f5687c78f-8jzmv   1/1     Running   0          134m   172.17.109.16   10.240.0.85   <none>           <none>
my-app-block-69956b9d94-snn5k        1/1     Running   0          26s    172.17.109.23   10.240.0.85   <none>           <none>

❯ kubectl exec my-app-block-69956b9d94-snn5k -it -- bash
bash-5.1# lsblk
bash: lsblk: command not found
bash-5.1# df -h
Filesystem                Size      Used Available Use% Mounted on
overlay                  97.3G      8.3G     84.7G   9% /
tmpfs                    64.0M         0     64.0M   0% /dev
tmpfs                     7.7G         0      7.7G   0% /sys/fs/cgroup
/dev/vdd                  9.7G     24.0K      9.7G   0% /myvolblock
/dev/vda2                97.3G      8.3G     84.7G   9% /etc/hosts
/dev/vda2                97.3G      8.3G     84.7G   9% /dev/termination-log
/dev/vda2                97.3G      8.3G     84.7G   9% /etc/hostname
/dev/vda2                97.3G      8.3G     84.7G   9% /etc/resolv.conf
shm                      64.0M         0     64.0M   0% /dev/shm
tmpfs                    12.8G     16.0K     12.8G   0% /run/secrets/kubernetes.io/serviceaccount
tmpfs                     7.7G         0      7.7G   0% /proc/acpi
tmpfs                    64.0M         0     64.0M   0% /proc/kcore
tmpfs                    64.0M         0     64.0M   0% /proc/keys
tmpfs                    64.0M         0     64.0M   0% /proc/timer_list
tmpfs                    64.0M         0     64.0M   0% /proc/sched_debug
tmpfs                     7.7G         0      7.7G   0% /proc/scsi
tmpfs                     7.7G         0      7.7G   0% /sys/firmware
bash-5.1# cd myvolblock/
bash-5.1# ls -la
total 24
drwxr-xr-x    3 root     root          4096 Aug 28 07:48 .
drwxr-xr-x    1 root     root          4096 Aug 28 07:48 ..
drwx------    2 root     root         16384 Aug 28 07:48 lost+found
bash-5.1# touch test
bash-5.1# ls -la
total 24
drwxr-xr-x    3 root     root          4096 Aug 28 07:51 .
drwxr-xr-x    1 root     root          4096 Aug 28 07:48 ..
drwx------    2 root     root         16384 Aug 28 07:48 lost+found
-rw-r--r--    1 root     root             0 Aug 28 07:51 test
bash-5.1#
```
- [ ] Expansion
```
❯ kubectl edit pvc my-pvc-block
persistentvolumeclaim/my-pvc-block edited
❯
❯ kubectl get pvc -w
NAME           STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS                VOLUMEATTRIBUTESCLASS   AGE
my-pvc-block   Bound    pvc-621d1283-e1fc-4d0c-ba63-a8533a2f700c   20Gi       RWO            ibmc-vpc-block-5iops-tier   <unset>                 4m28s
```
- [ ] Detach and delete PVC
```
❯ kubectl delete -f dep.yaml
deployment.apps "my-app-block" deleted
❯
❯
❯ kubectl delete -f pvc.yaml
persistentvolumeclaim "my-pvc-block" deleted
❯
❯ kubectl get pods
NAME                                 READY   STATUS    RESTARTS   AGE
auto-expansion-pod-f5687c78f-8jzmv   1/1     Running   0          139m
❯
❯ kubectl get pvc
No resources found in default namespace.
```
- [ ] Delete csi resources 
```
❯ kubectl get pods -n kube-system -o wide | grep block
❯
```

**Review by sameshai — APPROVED** [review]
/lgtm

**Comment by prankulmahajan** [comment]
### Environments to test:

- [ ] Kubernetes service with Ubuntu 24
- [ ] Openshift with RHEL 8.9 OS
- [ ] Openshift with RHEL 9.6 OS

### Test cases:
- [ ] New install
  - [ ] Resource status
  - [ ] PVC creation + Attachment + r/w
  - [ ] Expansion
  - [ ] Detachment + PVC deletion
- [ ] Patch install
  - [ ] Resource should be ready with existing PVC and attachment healthy
  - [ ] Create new attachement
  - [ ] Detachment + PVC deletion

File: pkg/ibmcsidriver/fileOps_test.go [hunk 1]
```diff
@@ -0,0 +1,126 @@
+/**
+ *
+ * Copyright 2024- IBM Inc. All rights reserved
```
Comment (arahamad) [inline pkg/ibmcsidriver/fileOps_test.go:3]: I don't think so you will add your corporate copyright
please check other existing files and accordingly add copyright

**Review by arahamad — CHANGES_REQUESTED** [review]
please change the copyright, other things looks good

can you also come up with release note with this changes

File: pkg/ibmcsidriver/fileOps_test.go (hunk 1 above)
Comment (prankulmahajan) [reply pkg/ibmcsidriver/fileOps_test.go]: Updated

**Review by arahamad — APPROVED** [review]
/lgtm

please verify changes also

**Pull request merged**